File: Java Spring Boot Project/CompanyDirectory/src/main/java/com/mindthecode/CompanyDirectory/models/responses/AllDepartmentsResponse.java
package com.mindthecode.CompanyDirectory.models.responses;

import java.util.List;

public class AllDepartmentsResponse {
    List<DepartmentResponse> departments;

    public AllDepartmentsResponse(List<DepartmentResponse> departments) {
        this.departments = departments;
    }

    public List<DepartmentResponse> getDepartments() {
        return departments;
    }

    public void setDepartments(List<DepartmentResponse> departments) {
        this.departments = departments;
    }

    @Override
    public String toString() {
        return "AllDepartmentsResponse{" +
                "departments=" + departments +
                '}';
    }
}

File: Java Spring Boot Project/company_directory_app/lib/models/responses/all_departments_response.dart
import 'package:company_directory_app/models/responses/department_response.dart';

class AllDepartmentsResponse {
  final List<DepartmentResponse> departments;

  AllDepartmentsResponse({this.departments});

  factory AllDepartmentsResponse.fromJson(Map<String, dynamic> json) {
    var list = json['departments'] as List;
    List<DepartmentResponse> itemsList = list.map((i) => DepartmentResponse.fromJson(i)).toList();

    return AllDepartmentsResponse(departments: itemsList);
  }
}
